### PR TITLE
Update `Swap` section

### LINKED ISSUE #94: When switching to Swap from other sections, error `Cannot exchange by requested amount` appears

### LINKED ISSUE #97: A react warning when using stepping control to change value in `TokenPicker`
Clicking on the stepping control

Screen Shot 2022-03-01 at 20 42 56

... cause this warning in the terminal

Screen Shot 2022-03-01 at 20 42 03

### PR DESCRIPTION
**Pull request by ken-centrality**
## Description

Overhaul the Swap section with updated UI elements / transitions effects

Closes #97 
Closes #94 

#### Changes

- Move `theme.ts` to the new `ThemeProvider` component
- Utilise built-in theming system to determine colour palette per section plus `text`, `background` colours and default `box-shadow`
- Switch to use `Roboto Mono` for numeric text
- Switch to use MUI Dialog to make use of its transition effects
- Switch to use MUI Select dropdown in `TokenInput` to make use of its transition effects
- Use native `form` to handle input validations
- Update `getStaticProps` in swap to use `fetchSwapAssets`
- Split the `pages/swap.tsx` into small components

#### Fixes

- Avoid showing raw errors when submitting the unexpected values
- Avoid hard-coding CPAY Asset ID when fetching gas estimate
- Avoid Info Panel only show when `balances` are fetched and there is input
- Fix `showDialog` doesn't work in asynchronous mode
- Fix issue where transaction errors are not surfaced up
- Fix issue where canceling the signing prompt doesn't close the "Swap in progress..." dialog 

#### Adds

- Add `ThemeProvider` and `CssGlobal`
- Add a lighter `TokenInput` component
- Add `big.js` to handle all arithmetics
- Add `decimalValues` to `CENNZAsset` and `EthereumToken` for convenience
- Add transition / animation effects
- Add `constants.ts` to type cast environment variables
- Add `useSwapGasFee` hook
- Add `useTokenFetcher` hook as shared utility to fetch required tokens
- Add `signAndSendTx` utility to send any extrinsic plus capture any errors that happen along the way
- Add `StandardButton` component
- Add `SubmitButton` component
- Add `SwapProvider` component

### CODE REVIEW

**Review by aidan-starke — CHANGES_REQUESTED** [review]
- can you chuck in a bit of whitespace between the blue and black text here?

Screen Shot 2022-03-11 at 9 00 44 AM

- can `utils/swap/index` be deleted?

**Review by eupston — APPROVED** [review]
looks really great! very clean code and definitely learned a thing or two

File: components/SwapStats.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+export default SwapStats;
+
+const styles = {
+	root: ({ palette }: Theme) => css`
+		margin-top: 2em;
+		padding: 1.5em;
+		color: ${palette.text.primary};
+		background-color: ${palette.background.main};
+		position: relative;
+
```
Comment (eupston) [inline components/SwapStats.tsx:94]: might be worth just chucking a border radius here to match styling of other components

**Comment by ken-centrality** [comment]
Ready for round 2 @aidan-starke @eupston

**Review by aidan-starke — APPROVED** [review]

**Comment by aidan-starke** [comment]
lgtm! might want to change the `^` for `SETTINGS` as well

**Comment by ken-centrality** [comment]
> lgtm! might want to change the `^` for `SETTINGS` as well

🤦‍♂️ oh duh! it's up now, thanks!

**Pull request merged**